### ISSUE TITLE
zuul does nothing when using `--phantom` flag

### ISSUE THREAD

**Issue by Fauntleroy** [defunctzombie/zuul#222]
When I try to run zuul with the `--phantom` flag it does nothing (and I eventually just have to ctrl+c out). Here's what that looks like:

```
$ npm test

> react-simple-dropdown@0.1.0 test /Users/tkempf/Repos/react-simple-dropdown
> zuul --debug --phantom -- test/**/*.{js,jsx}

^C
```

I had a suspicion that maybe running zuul through an npm script might be causing an issue, so I installed zuul globally and tried again:

```
$ zuul --debug --phantom -- test/**/*.{js,jsx}

module.js:338
    throw err;
    ^
Function.Module._resolveFilename (module.js:336:15),Function.Module._load (module.js:278:25),Module.require (module.js:365:17),require (module.js:384:17),PhantomBrowser.start (/usr/local/lib/node_modules/zuul/lib/PhantomBrowser.js:28:21),/usr/local/lib/node_modules/zuul/lib/zuul.js:113:28,Server.<anonymous> (/usr/local/lib/node_modules/zuul/lib/zuul.js:58:9),Server.g (events.js:199:16)
```

I'd installed phantomjs in my project directory, but it just wasn't working. So in an attempt to debug further I installed phantomjs in my global install of zuul and tried again:

```
$ zuul --debug --phantom -- test/**/*.{js,jsx}
^C
```

I'm not sure what it is, but zuul *really* doesn't want to work with phantomjs for me. The code for the project I'm working on can be found here: https://github.com/Fauntleroy/react-simple-dropdown

**Comment by vvo** [comment]
1. does --local works?
2. the issue with phantomjs is that uncaught errors are not shown in the console, thus if your code is not phantomjs compatible then you are screwed, nothing will be shown and it hangs.

What you could try is launch in local mode, see if a regular browser works, then try to open a phantomjs command line and navigate to the page.

thanks

**Comment by revolunet** [comment]
same here;

 - `--local`: i can run tests in browser
 - `--phantom` : nothing happens, with either successful tests or not

Any way to get a simple CLI exit code from zuul ?

**Comment by revolunet** [comment]
mmm after testing in phantom, in my case looks like Function.prototype.bind is not available in PhantomJS (https://groups.google.com/forum/#!msg/phantomjs/r0hPOmnCUpc/uxusqsl2LNoJ)

```
Status: success
TypeError: 'undefined' is not a function (evaluating 'RegExp.prototype.test.bind(/^(data|aria)-[a-z_][a-z\d_.\-]*$/)')

  http://localhost:3100/__zuul/test-bundle.js:10872
  http://localhost:3100/__zuul/test-bundle.js:20 in __webpack_require__
  ...
  http://localhost:3100/__zuul/test-bundle.js:21184
```

Is there any way to make zuul open a browser then close when test suite ends of fails and return correct exit status ?

**Comment by vvo** [comment]
Yes this function.prototype.bind not available already made the whole dev community waste hundreds of dev years :D

From my experience I never managed to make phantomjs fail/fail my test when the tests are failing/uncaught.

What I do is that if the phantomjs test fails then I will just open a real browser (npm run dev) and see what's wrong/debug.

I think you have the same issue by just using mocha + phantomjs (without zuul or anything).

**Comment by vvo** [comment]
> Is there any way to make zuul open a browser then close when test suite ends of fails and return correct exit status ?

Maybe, did not overlook it
